### PR TITLE
Parameters with study import from existing case

### PR DESCRIPTION
**Pull request by klesaulnier**
Signed-off-by: LE SAULNIER Kevin <kevin.lesaulnier@rte-france.com>

### CODE REVIEW

File: src/components/dialogs/create-study-dialog.js [hunk 1]
```diff
@@ -151,14 +158,39 @@ export const CreateStudyDialog = ({ open, onClose, providedCase }) => {
...
+    const [currentParameters, paramsComponent, resetImportParamsToDefault] =
+        useMeta(metasAsArray);
+
     //Inits the dialog
     useEffect(() => {
         if (open && providedCase) {
             setStudyName(providedCase?.elementName);
             setCaseExist(true);
             dispatch(selectCase(providedCase?.elementUuid));
+            getCaseImportParameters(providedCase?.elementUuid).then(
```
Comment (SlimaneAmar) [inline src/components/dialogs/create-study-dialog.js]: Why `providedCase?` ?
`providedCase` never null in the block ?

File: src/components/dialogs/create-study-dialog.js [hunk 2]
```diff
@@ -415,47 +478,74 @@ export const CreateStudyDialog = ({ open, onClose, providedCase }) => {
...
-                        </div>
+                                <Stack
+                                    marginTop="0.7em"
+                                    direction="row"
+                                    justifyContent="space-between"
+                                    alignItems="center"
+                                >
+                                    {MakeAdvancedParameterButton(
+                                        isParamsDisplayed,
+                                        intl.formatMessage({
```
Comment (SlimaneAmar) [inline src/components/dialogs/create-study-dialog.js]: Only `importParameters`
Remove ` intl.formatMessage`

File: src/components/dialogs/create-study-dialog.js [hunk 3]
```diff
@@ -415,47 +478,74 @@ export const CreateStudyDialog = ({ open, onClose, providedCase }) => {
...
-                                })}
-                                validationButtonText={intl.formatMessage({
-                                    id: 'confirmDirectoryDialog',
-                                })}
-                                contentText={intl.formatMessage({
-                                    id: 'moveItemContentText',
-                                })}
-                            />
-                        </div>
+                                <Stack
```
Comment (SlimaneAmar) [inline src/components/dialogs/create-study-dialog.js]: Why this component ?

File: src/components/dialogs/create-study-dialog.js [hunk 4]
```diff
@@ -151,14 +158,39 @@ export const CreateStudyDialog = ({ open, onClose, providedCase }) => {
...
+
     const [triggerReset, setTriggerReset] = React.useState(true);
 
+    const [isParamsDisplayed, setIsParamsDisplayed] = useState(false);
+
+    const handleShowParametersClick = () => {
+        setIsParamsDisplayed((oldValue) => !oldValue);
+    };
+
+    const metasAsArray = formatWithParameters?.parameters || [];
```
Comment (SlimaneAmar) [inline src/components/dialogs/create-study-dialog.js]: Why we do not use directly the state  formatWithParameters with the final value (parameters or [])

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/utils/rest-api.js [hunk 5]
```diff
@@ -788,3 +791,16 @@ export function fetchPath(elementUuid) {
             : response.text().then((text) => Promise.reject(text))
     );
 }
+
+export function getCaseImportParameters(caseUuid) {
+    console.info(`get import parameters for case '${caseUuid}' ...`);
```
Comment (SlimaneAmar) [inline src/utils/rest-api.js:798]: Why we do not call directly `study-server `?

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/components/dialogs/create-study-dialog.js (hunk 1 above)
Comment (klesaulnier) [reply src/components/dialogs/create-study-dialog.js]: I agree, it was to "copy" the previous lines behaviour. Fixed

File: src/components/dialogs/create-study-dialog.js (hunk 2 above)
Comment (klesaulnier) [reply src/components/dialogs/create-study-dialog.js]: fixed

File: src/components/dialogs/create-study-dialog.js (hunk 3 above)
Comment (klesaulnier) [reply src/components/dialogs/create-study-dialog.js]: it's useless now, removed

File: src/components/dialogs/create-study-dialog.js (hunk 4 above)
Comment (klesaulnier) [reply src/components/dialogs/create-study-dialog.js]: fixed, I agree it makes the component easier to read

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**